projects/packages/videopress/src/client/admin/components/input/types.ts

```diff
@@ -1,40 +1,54 @@
 import type React from 'react';
 
-export type InputProps = {
+type InputBaseProps = {
 	/**
 	 * Optional classname to apply to the root element.
 	 */
 	className?: string;
 
 	/**
-	 * The input value.
+	 * Input size.
 	 */
-	value?: string;
+	size?: 'small' | 'large';
 
 	/**
-	 * The input placeholder.
+	 * Input label.
 	 */
-	placeholder?: string;
+	label?: React.ReactNode;
 
 	/**
-	 * Whether the input is disabled.
+	 * Callback to be invoked when the input value changes.
 	 */
-	disabled?: boolean;
+	onChange: ( value: string ) => unknown;
 
+	/**
+	 * Callback to be invoked when the user presses the Enter key.
+	 */
+	onEnter: ( value: string ) => unknown;
+};
+
+type Input = Omit< React.InputHTMLAttributes< HTMLInputElement >, 'size' > & {
 	/**
 	 * Optional icon.
 	 */
 	icon?: React.ReactNode;
-
 	/**
-	 * Callback to be invoked when the input value changes.
+	 * Input types.
 	 */
-	onChange: ( value: string ) => unknown;
+	type?: 'text' | 'password' | 'email' | 'number' | 'search' | 'tel' | 'url';
+} & InputBaseProps;
 
+type TextArea = React.TextareaHTMLAttributes< HTMLTextAreaElement > & {
 	/**
-	 * Callback to be invoked when the user presses the Enter key.
+	 * No support for icon when using textarea.
 	 */
-	onEnter: ( value: string ) => unknown;
-};
+	icon?: undefined;
```

Comment:
Maybe it is better to not declare `icon`?

Reply:
Nope. We need to declare as undefined to Typescript understand that this property doesn't exist at this moment

Reply:
Ah, I see now in the `InputProps`.